### PR TITLE
ISPN-7779 State transfer does not work with protobuf encoded entities

### PR DESCRIPTION
**Pull request by anistor**
* we need to ensure __protobuf_metadata cache is started and ST complete before any cache using protobuf gets started

https://issues.jboss.org/browse/ISPN-7779

### CODE REVIEW

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java [hunk 1]
```diff
@@ -131,6 +134,15 @@ public void cacheStarting(ComponentRegistry cr, Configuration cfg, String cacheN
       }
    }
 
+   private void addProtobufMetadataCacheDependency(GlobalComponentRegistry gcr, String cacheName) {
+      ProtobufMetadataManagerImpl protobufMetadataManager = (ProtobufMetadataManagerImpl) gcr.getComponent(ProtobufMetadataManager.class);
+      protobufMetadataManager.getCache(); // force the cache to start
```
Comment (gustavonalle) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/LifecycleManager.java]: You could use ``` EmbeddedCacheManager#startCaches(protobufMetadataManager)``` and spare the comment :)

**Comment by gustavonalle** [comment]
LGTM

**Comment by anistor** [comment]
Updated.

**Comment by gustavonalle** [comment]
integrated, thanks!

**Pull request merged**